src/common/resources/application.ts

```diff
@@ -0,0 +1,56 @@
+import { Group, Application, User } from 'common/typedefs';
+import { IListParams, IListResponse } from 'common/typedefs/Resource';
+import {
+  addApplicationToGroup,
+  addApplicationToUser,
+  createApplication,
+  deleteApplication,
+  getApp,
+  getApps,
+  removeApplicationFromGroup,
+  removeApplicationFromUser,
+  updateApplication,
+} from 'services';
+
+interface ApplicationResourceInterface {
+  createItem: ({ item }: { item: Partial<Application> }) => Promise<Application>;
+  deleteItem: ({ item }: { item: Application }) => Promise<string>;
+  updateItem: ({ item }: { item: Application }) => Promise<Application>;
+  getList: (params: IListParams) => Promise<IListResponse>;
+  getChildList?: (
+    resourceType: string,
+    childResourceType: string,
+    id: string,
+  ) => Promise<IListResponse>;
```

Comment:
I know you havent implemented this yet so it may be good timing to leave a comment on this interface...

this is `applicationInterface.getChildList` so we definitely don't need the `resourceType` as an argument. Maybe we want to reorganize it like:

```
interface ApplicationResourceInterface {
  ... stuff
  getChild: {
    groups: (applicationId: string) => Promise<IListResponse<Group>>;
    users: (applicationId: string) => Promise<IListResponse<User>>;
    ... etc
    }
}
```

Reply:
This interface change probably also applies to other resources...